### PR TITLE
Feat: ECS compatibility

### LINKED ISSUE #6: ECS compatibility
ECS will provide fields for syslog priority / severity  facility soon, see https://github.com/elastic/ecs/pull/525

It would be nice if this plugin could be updated, so the correct ECS fields are used. Currently we need to rename the fields like this:

```
    syslog_pri { } 
    mutate {
        rename => { "syslog_pri" => "log.priority" }
        rename => { "syslog_facility" => "log.facility.name" }
        rename => { "syslog_facility_code" => "log.facility.code" }
        rename => { "syslog_severity" => "log.level" }
        rename => { "syslog_severity_code" => "log.severity" }
    }
```

Thanks for looking into this!

### PR DESCRIPTION
**Pull request by kares**
**Elastic Common Schema Support**

using `log.syslog.` field names as prescribed by the schema

---

closing https://github.com/logstash-plugins/logstash-filter-syslog_pri/issues/8
resolves https://github.com/logstash-plugins/logstash-filter-syslog_pri/issues/6

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.1.0
+  - Feat: ECS compatiblity [#9](https://github.com/logstash-plugins/logstash-filter-syslog_pri/pull/9)
```
Comment (karenzone) [inline CHANGELOG.md]:
```suggestion
  - Feat: ECS compatibility [#9](https://github.com/logstash-plugins/logstash-filter-syslog_pri/pull/9) 
```

Comment (karenzone) [reply CHANGELOG.md]: Please fix the spelling of "compatibility" in the PR title and changelog (because we link to them from release notes).  :-)

**Review by karenzone — COMMENTED** [review]
Doc review: Docs look good. Thanks for keeping up the consistency. I noted a minor spelling error in the changelog and PR title.

File: spec/filters/syslog_pri_spec.rb [hunk 2]
```diff
@@ -5,109 +5,173 @@
...
+        end
+
+        it "default syslog_facility is user-level" do
+          subject.filter(event)
+          if ecs_compatibility?
+            expect(event.get("[log][syslog][facility][name]")).to eq("local1")
+            expect(event.get("[log][syslog][facility][code]")).to eq(17)
+          else
+            expect(event.get("syslog_facility")).to eq("local1")
+            expect(event.get("syslog_facility_code")).to eq(17)
```
Comment (kaisecheng) [inline spec/filters/syslog_pri_spec.rb:170]: can you use `ecs_select` to simplify the condition?
```suggestion
          expect( event.get ecs_select[disabled: "syslog_facility", v1: "[log][syslog][facility][name]"] ). to eq("local1")
          expect( event.get ecs_select[disabled: "syslog_facility_code", v1: "[log][syslog][facility][code]"] ). to eq("17")
```

**Review by kaisecheng — APPROVED** [review]
This PR looks good :four_leaf_clover:
I left a comment on the test to use `ecs_select` potentially

File: spec/filters/syslog_pri_spec.rb (hunk 2 above)
Comment (kares) [reply spec/filters/syslog_pri_spec.rb:170]: Thanks Kaise, we could do smt similar I feel like there's too much "meta-fu" in the specs already.
So I prefer to keep the ecs vs legacy asserts explicit, also `ecs_select` is an internal method to the plugin (we would need to use something else).

**Comment by roaksoax** [comment]
@kares shouldn't we be documenting all of the fields that get mapped to ECS schema? @yaauie

**Comment by kares** [comment]
not sure, should we? the syslog ECS fields are unambiguous and we do not document every field plugin is using ...
although an example "legacy" vs ECS might be worth the efforts.

**Pull request merged**